lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -407,17 +406,20 @@ def normal_void(authorization, options)
         end
       end
 
-      def add_payment_source(xml, source, options, action = nil)
-        return unless source
+      def add_payment_method(xml, payment_method, options, action = nil)
+        return unless payment_method
 
-        if source.is_a?(String)
-          add_token_payment_method(xml, source, options)
-        elsif card_brand(source) == 'check'
-          add_check(xml, source)
-        elsif card_brand(source) == 'apple_pay'
```

Comment:
Auth.net may not care about the `apple_pay` flagging, but I'm wondering if it's still worth it to keep this check in our decision tree. It will make a difference to us if we ever need to add GooglePay or network tokenization outside of digital wallets 🤔

Reply:
This doesn't really work because it won't ever return `apple_pay` but the returns the card brand for the PM. In my understanding this works for any Tokenized Credit Card because their docs doesn't say that this is only for ApplePay.